fastqpreprocessing/src/utilities.cpp

```diff
@@ -0,0 +1,308 @@
+          {"white-list",    required_argument, 0, 'w'},
+          {0, 0, 0, 0}
+        };
+
+      // help messages when the user types -h
+      const char *help_messages[] = {
+           "verbose messages  ",
+           "barcode length [required]",
+           "UMI length [required]",
+           "output BAM file in GB [optional: default 1 GB]",
+           "sample id [required]",
+           "I1",
+           "R1 [required]",
+           "R2 [required]",
+           "whitelist (from cellranger) of barcodes [required]",
+      };
+
+
+      /* getopt_long stores the option index here. */
+      int option_index = 0;
+
+      c = getopt_long(argc, argv, "b:u:B:s:I:R:r:w:", \
+                      long_options, &option_index);
+
+      /* Detect the end of the options. */
+      if (c == -1)
+        break;
+
+      // process the option or arguments
+      switch (c) {
+        case 0:
+          /* If this option set a flag, do nothing else now. */
+            if (long_options[option_index].flag != 0)
+                break;
+            printf("option %s", long_options[option_index].name);
+            if (optarg)
+                printf(" with arg %s", optarg);
+            printf("\n");
+            break;
+        case 'b':
+            options.barcode_length = atoi(optarg);
+            break;
+        case 'u':
+            options.umi_length = atoi(optarg);
+            break;
+        case 'B':
+            options.bam_size = atof(optarg);
+            break;
+        case 's':
+            options.sample_id = string(optarg);
+            break;
+        case 'I':
+            options.I1s.push_back(string(optarg));
+            break;
+        case 'R':
+            options.R1s.push_back(string(optarg));
+            break;
+        case 'r':
+            options.R2s.push_back(string(optarg));
+            break;
+        case 'w':
+            options.white_list_file = string(optarg);
+            break;
+        case '?':
+        case 'h':
+          i = 0;
+          printf("Usage: %s [options] \n", argv[0]);
+          while (long_options[i].name != 0) {
+             printf("\t--%-20s  %-25s  %-35s\n", long_options[i].name, \
+                     long_options[i].has_arg == no_argument? \
+                     "no argument" : "required_argument", \
+                     help_messages[i]);
+             i = i + 1;
+          }
+          /* getopt_long already printed an error message. */
+          return;
+          break;
+        default:
+          abort();
+        }
+    }
+
+  // Check the options
+
+  // number of R1 and R2 files should be equal
+  if ((options.R1s.size() != options.R2s.size()) || (options.R1s.size() == 0)) {
+     std::cout << "ERROR: Unequal number of R1 and R2 fastq files in input\n";
+     std::cerr << "ERROR: Unequal number of R1 and R2 fastq files in input\n";
+     exit(1);
+  }
+
+  if ((options.I1s.size() != options.R1s.size()) and (options.I1s.size() != 0)) {
```

Comment:
even though `and` is the same as `&&` here, for consistency I prefer `&&`
```suggestion
  if ((options.I1s.size() != options.R1s.size()) && (options.I1s.size() != 0)) {
```

Reply:
changed to &&